### ISSUE TITLE
Run Keyword If with ELSE and dictionaries.

### ISSUE THREAD

**Issue by Mario-P** [nokia/RED#96]
In following test I have an RED validation error message in place where everything is ok.
```
*** Variables ***
&{dict}    
${two}    
${one}    

*** Test Cases ***
Test
    Run Keyword If    ${one}==${two}    My Keyword    &{dict}    ELSE    My other keyword    &{dict}

*** Keywords ***
My Keyword
    [Arguments]    &{my_dict}
    Log Many    &{my_dict}
    
My other keyword
    [Arguments]    &{my_dict}
    Log Many    &{my_dict}
```
Validation error is in line 8 (Run Keyword If    ${one}==${two}    My Keyword    &{dict}    ELSE    My other keyword    &{dict}) and says: "Positional arguments cannot be used after named arguments".

I guess this is about using ELSE after dict but in this case it is perfectly fine.

**Comment by michalanglart** [comment]
Hi, 

This is loosely related to #28. The problem here is that arguments for Run Keyword If and other keywords from this family are treated in a special way. RED on the other hand validates those arguments like if they would be provided normal way.

We need to take a look, there is no workaround for this as for now. You may change reported severity of this validation error in RED preferences, however all such problems will be reported with lowered severity.

**Comment by michalanglart** [comment]
Fixed in 0.8.6. 

So-called run-keyword-variants from BuiltIn library now have arguments validated differently than casual keywords
